packages/rspack/src/migrations/update-22-0-0/remove-deprecated-options.spec.ts

```diff
@@ -0,0 +1,431 @@
+              options: {
+                outputPath: 'dist/apps/test-app',
+                main: 'apps/test-app/src/main.ts',
+                deleteOutputPath: true,
+                tsConfig: 'apps/test-app/tsconfig.app.json',
+              },
+            },
+          },
+        })
+      );
+
+      // ACT
+      await removeDeprecatedOptions(tree);
+
+      // ASSERT
+      const projectJson = JSON.parse(
+        tree.read('apps/test-app/project.json', 'utf-8')
+      );
+      expect(
+        projectJson.targets.build.options.deleteOutputPath
+      ).toBeUndefined();
+      expect(projectJson.targets.build.options.outputPath).toBe(
+        'dist/apps/test-app'
+      );
+      expect(projectJson.targets.build.options.main).toBe(
+        'apps/test-app/src/main.ts'
+      );
+    });
+
+    it('should remove deleteOutputPath from project.json configurations', async () => {
+      // ARRANGE
+      tree.write(
+        'apps/test-app/project.json',
+        JSON.stringify({
+          name: 'test-app',
+          root: 'apps/test-app',
+          sourceRoot: 'apps/test-app/src',
+          projectType: 'application',
+          targets: {
+            build: {
+              executor: '@nx/rspack:rspack',
+              options: {
+                outputPath: 'dist/apps/test-app',
+                main: 'apps/test-app/src/main.ts',
+              },
+              configurations: {
+                production: {
+                  deleteOutputPath: false,
+                  optimization: true,
+                },
+              },
+            },
+          },
+        })
+      );
+
+      // ACT
+      await removeDeprecatedOptions(tree);
+
+      // ASSERT
+      const projectJson = JSON.parse(
+        tree.read('apps/test-app/project.json', 'utf-8')
+      );
+      expect(
+        projectJson.targets.build.configurations.production.deleteOutputPath
+      ).toBeUndefined();
+      expect(
+        projectJson.targets.build.configurations.production.optimization
+      ).toBe(true);
+    });
+
+    it('should add comment when removing from rspack config files', async () => {
+      // ARRANGE
+      tree.write(
+        'apps/test-app/project.json',
+        JSON.stringify({
+          name: 'test-app',
+          root: 'apps/test-app',
+          sourceRoot: 'apps/test-app/src',
+          projectType: 'application',
+          targets: {
+            build: {
+              executor: '@nx/rspack:rspack',
+              options: {},
+            },
+          },
+        })
+      );
+      tree.write(
+        'apps/test-app/rspack.config.ts',
+        `export default {
+  entry: './src/main.ts',
+  deleteOutputPath: true,
+  output: {
+    path: 'dist',
+  },
+};`
+      );
+
+      // ACT
+      await removeDeprecatedOptions(tree);
+
+      // ASSERT
+      const configContent = tree.read(
+        'apps/test-app/rspack.config.ts',
+        'utf-8'
+      );
+      expect(configContent).toContain(
```

Comment:
Nit: if we use inline snapshots with the full config it would be easier to see that it is valid after update.

Reply:
+1